src/Raven.Server/ServerWide/DebugInfo/ServerEndpointDebugInfoDataSource.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Http;
+using Raven.Server.Routing;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.ServerWide.DebugInfo
+{
+    public class ServerEndpointDebugInfoDataSource : IDebugInfoDataSource
+    {
+        private readonly RouteInformation _routeInformation;
+        private readonly RequestExecutor _requestExecutor;
+        private readonly CancellationToken _token;
+
+        public ServerEndpointDebugInfoDataSource(string url, RouteInformation routeInformation, CancellationToken token)
+        {
+            _requestExecutor = RequestExecutor.CreateForSingleNode(url, null, null);        
+            _routeInformation = routeInformation;
+            _token = token;
+        }
+
+        public string FullPath
+        {
+            get
+            {
+                var filename = _routeInformation.Path;
+                if (filename.StartsWith("debug/", StringComparison.OrdinalIgnoreCase))
+                    filename = filename.Replace("debug/", string.Empty);
+                if (filename.StartsWith("/debug/", StringComparison.OrdinalIgnoreCase))
+                    filename = filename.Replace("/debug/", string.Empty);
+                filename = filename.Replace("/", ".");
+
+                return $"Server-wide\\{filename}.json";
+            }
+        }
+
+        public async Task<BlittableJsonReaderObject> GetData(JsonOperationContext context)
+        {
+            var cmd = new GetRawResponseCommand(_routeInformation.Method, _routeInformation.Path);           
```

Comment:
Following up on our conversation, we'll use a single endpoint that will internally call the relevant stuff to generate a zip file.
For cluster, we'll have an endpoint that will hit all the cluster nodes and generate a zip file with an internal file for each node (`A.zip`, `B.zip`, etc)